### ISSUE TITLE
Restore originally installed version after tav run

### ISSUE THREAD

**Issue by torfsen** [watson/test-all-versions#4]
After running `tav`, the targeted dependency is still installed in the last version checked by `tav`:

1. `npm install some-dependency@2`
2. `tav some-dependency "^1.0.0" npm test`
3. `some-dependency` is now installed in version 1 in `node_modules`

This is very confusing, especially if `tav` exits early after a failed test. I suggest that `tav` automatically restores the originally installed version before it exits.

**Comment by watson** [comment]
Good idea. In many cases I actually like having it installed, so if I have a failure in the middle of the test, I can start debugging right away because the dependency is still installed. But it might make sense to have a flag that either enables or disables this functionality.

I'm not sure however how to go about doing this as the installed dependencies potentially will overwrite other dependencies already installed in `node_modules`. So to restore the state of the `node_modules` directory till after this command was run might be tricky.

More research is needed 😅
